### PR TITLE
Email

### PR DESCRIPTION
**Pull request by jiwonkim19**
Hi. Take a look at the serializers,py in requestpage folder. Once a request is submitted, the create() method in RequestSerializer is overridden.

### CODE REVIEW

File: requestpage/views.py [hunk 1]
```diff
@@ -1,20 +1,24 @@
...
 from .serializers import RequestSerializer
 from .models import Request
 from django.http import HttpResponse, JsonResponse
 from django.views.generic import View
 from django.conf import settings
-from django.views.generic import View 
+from django.core.mail import send_mail
 import os 
-from django.middleware.csrf import get_token
+# import pdb; pdb.set_trace()
```
Comment (eternal44) [inline requestpage/views.py:13]: You typically want to remove code you've used for debugging. It's handy to have around as a quick reference but can be distracting for other readers of your code.

File: requestpage/serializers.py [hunk 2]
```diff
@@ -1,7 +1,26 @@
...
 
 class RequestSerializer(serializers.ModelSerializer):
     class Meta:
         model = Request
-        fields = ('id', 'user', 'reqType', 'reqText', 'status', 'updated_at', 'created_at')
\ No newline at end of file
+        fields = ('id', 'user', 'reqType', 'reqText', 'status', 'updated_at', 'created_at')
+    
+    def create(self, validated_data):
+      instance = super(RequestSerializer, self).create(validated_data)
```
Comment (eternal44) [inline requestpage/serializers.py:14]: I'd strongly recommend renaming the variable to reflect whatever it's an instance of. Or, is this a common pattern in django apps that I'm not aware of?

File: requestpage/views.py [hunk 3]
```diff
@@ -1,20 +1,24 @@
 from rest_framework import viewsets
+from rest_framework.parsers import JSONParser
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from rest_framework.decorators import action
```
Comment (eternal44) [inline requestpage/views.py:5]: I see you've imported functions and classes but I don't see them being used anywhere. Does Django do something with these imports or do you have to do something with them?

File: joytechapp/settings.py [hunk 4]
```diff
@@ -164,7 +165,17 @@
...
+EMAIL_BACKEND = env('EMAIL_BACKEND')
+EMAIL_HOST = env('EMAIL_HOST')
+EMAIL_PORT = env.int('EMAIL_PORT')
+EMAIL_USE_TLS = env.bool('EMAIL_USE_TLS')
+EMAIL_HOST_USER = env('EMAIL_HOST_USER')
+EMAIL_HOST_PASSWORD = env('EMAIL_HOST_PASSWORD')
+RECIPIENT_ADDRESS = env('RECIPIENT_ADDRESS')
+
+# ADMIN EMAIL
+ADMIN_EMAIL = env('ADMIN_EMAIL')
```
Comment (eternal44) [inline joytechapp/settings.py:178]: Can you update the README file? It'd be nice to know how you set up the email server so we can do the same on our laptops.

File: requestpage/serializers.py (hunk 2 above)
Comment (jiwonkim19) [reply requestpage/serializers.py:14]: Yep. it's a common pattern in django rest-framework for saving instances.

File: requestpage/views.py (hunk 3 above)
Comment (jiwonkim19) [reply requestpage/views.py:5]: Cleaned it up! Thanks!

File: requestpage/serializers.py (hunk 2 above)
Comment (eternal44) [reply requestpage/serializers.py:14]: Oh, I don't mean to call attention to creating a variable - just what to name the variable. In other words, is this instance a request? If so, how about naming this variable `request`?

File: src/Components/auth/Signup.js [hunk 5]
```diff
@@ -15,13 +17,30 @@ const Signup = () => {
...
+  const validateEmail = () =>
+    fetch(`/api/userprofile/?email=${email}`, {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json'
+      }
+    })
+    .then(res=>(res.json())
+    .then(data => {
+        if (data[0].email === email) {
```
Comment (jiwonkim19) [inline src/Components/auth/Signup.js:29]: `data[0] && data[0].email`

Comment (jiwonkim19) [reply src/Components/auth/Signup.js]: Done!

File: joytechapp/settings.py (hunk 4 above)
Comment (jiwonkim19) [reply joytechapp/settings.py]: I updated most of it but will need to include instructions on creating Google "app password (16-digit passcode that gives a less secure app or device permission to access your Google Account.)

**Pull request merged**